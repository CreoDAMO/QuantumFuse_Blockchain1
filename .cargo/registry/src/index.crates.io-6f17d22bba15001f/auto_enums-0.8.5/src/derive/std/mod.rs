// SPDX-License-Identifier: Apache-2.0 OR MIT

/// `std::error`
pub(crate) mod error;
/// `std::io`
pub(crate) mod io;
