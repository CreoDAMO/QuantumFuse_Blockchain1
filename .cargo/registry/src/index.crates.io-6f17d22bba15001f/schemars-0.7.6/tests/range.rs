mod util;
use schemars::JsonSchema;
use std::ops::{Bound, Range, RangeInclusive};
use util::*;

#[derive(Debug, JsonSchema)]
struct MyStruct {
    range: Range<usize>,
    inclusive: RangeInclusive<f64>,
    bound: Bound<String>,
}

#[test]
fn result() -> TestResult {
    test_default_generated_schema::<MyStruct>("range")
}
