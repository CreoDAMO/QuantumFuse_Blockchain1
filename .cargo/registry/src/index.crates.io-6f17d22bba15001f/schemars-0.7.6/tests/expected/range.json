{
  "$schema": "http://json-schema.org/draft-07/schema#",
  "title": "MyStruct",
  "type": "object",
  "required": [
    "bound",
    "inclusive",
    "range"
  ],
  "properties": {
    "bound": {
      "$ref": "#/definitions/Bound_of_String"
    },
    "inclusive": {
      "$ref": "#/definitions/Range_of_double"
    },
    "range": {
      "$ref": "#/definitions/Range_of_uint"
    }
  },
  "definitions": {
    "Bound_of_String": {
      "oneOf": [
        {
          "type": "object",
          "required": [
            "Included"
          ],
          "properties": {
            "Included": {
              "type": "string"
            }
          }
        },
        {
          "type": "object",
          "required": [
            "Excluded"
          ],
          "properties": {
            "Excluded": {
              "type": "string"
            }
          }
        },
        {
          "type": "string",
          "const": "Unbounded"
        }
      ]
    },
    "Range_of_double": {
      "type": "object",
      "required": [
        "end",
        "start"
      ],
      "properties": {
        "end": {
          "type": "number",
          "format": "double"
        },
        "start": {
          "type": "number",
          "format": "double"
        }
      }
    },
    "Range_of_uint": {
      "type": "object",
      "required": [
        "end",
        "start"
      ],
      "properties": {
        "end": {
          "type": "integer",
          "format": "uint",
          "minimum": 0.0
        },
        "start": {
          "type": "integer",
          "format": "uint",
          "minimum": 0.0
        }
      }
    }
  }
}